static/js/user_groups.ts

```diff
@@ -67,33 +93,41 @@ export function is_member_of(user_group_id, user_id) {
     return user_group.members.has(user_id);
 }
 
-export function add_members(user_group_id, user_ids) {
+export function add_members(user_group_id: number, user_ids: number[]): void {
     const user_group = user_group_by_id_dict.get(user_group_id);
+    if (user_group === undefined) {
+        blueslip.error("Could not find user group with ID " + user_group_id);
+        return;
+    }
 
     for (const user_id of user_ids) {
         user_group.members.add(user_id);
     }
 }
 
-export function remove_members(user_group_id, user_ids) {
+export function remove_members(user_group_id: number, user_ids: number[]): void {
     const user_group = user_group_by_id_dict.get(user_group_id);
+    if (user_group === undefined) {
+        blueslip.error("Could not find user group with ID " + user_group_id);
+        return;
+    }
 
     for (const user_id of user_ids) {
         user_group.members.delete(user_id);
     }
 }
 
-export function initialize(params) {
+export function initialize(params: {realm_user_groups: UserGroupRaw[]}): void {
     for (const user_group of params.realm_user_groups) {
         add(user_group);
     }
 }
 
-export function is_user_group(item) {
+export function is_user_group(item: Record<string, unknown>): item is UserGroup {
```

Comment:
Let's at least add a comment that (1) this function is a bad pattern and (2) this is used in pill contexts where the item should be a union of User, UserGroup, Stream.

Reply:
We should *annotate* it as accepting `User | UserGroup | Stream`, not just comment it. It would be unsound to call `is_user_group({members: "foo"})`, and we shouldn’t allow that to type check.

Reply:
Annotated `item` as `User | UserGroup | Stream`.